atlasdb-config/src/test/java/com/palantir/atlasdb/http/AtlasDbHttpProtocolVersionTest.java

```diff
@@ -17,83 +17,33 @@
 package com.palantir.atlasdb.http;
 
 import static org.assertj.core.api.Assertions.assertThat;
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.when;
 
-import javax.ws.rs.core.HttpHeaders;
+import java.util.Arrays;
+import java.util.Optional;
 
 import org.junit.Test;
 
-import com.google.common.collect.ImmutableList;
-import com.palantir.conjure.java.api.config.service.UserAgent;
-import com.palantir.conjure.java.api.config.service.UserAgents;
-
 public class AtlasDbHttpProtocolVersionTest {
-    private static final UserAgent USER_AGENT_1 = createBaseUserAgent("Bond", "0.0.7");
-    private static final UserAgent USER_AGENT_2 = createBaseUserAgent("Smith", "1.2.3");
-
-    private static final UserAgent.Agent AGENT_1 = UserAgent.Agent.of("Shield", "24.7.0");
-    private static final UserAgent.Agent CONJURE_AGENT = UserAgent.Agent.of(UserAgents.CONJURE_AGENT_NAME, "6.6.6");
-
-    private final HttpHeaders httpHeaders = mock(HttpHeaders.class);
-
     @Test
-    public void detectsConjureJavaRuntimeAgent() {
-        when(httpHeaders.getRequestHeader(HttpHeaders.USER_AGENT)).thenReturn(
-                ImmutableList.of(UserAgents.format(USER_AGENT_1.addAgent(CONJURE_AGENT))));
-
-        assertThat(AtlasDbHttpProtocolVersion.inferFromHttpHeaders(httpHeaders))
-                .isEqualTo(AtlasDbHttpProtocolVersion.CONJURE_JAVA_RUNTIME);
+    public void infersFromKnownVersionStrings() {
+        Arrays.stream(AtlasDbHttpProtocolVersion.values())
```

Comment:
would prefer if this was written out explicitly for each `AtlasDbHttpProtocolVersion`

Reply:
Sure, makes sense. Will fix